concourse/client/model.py

```diff
@@ -221,6 +252,24 @@ def find_tid(p):
                             return task_id
         return find_tid(plan)
 
+    def contains_version_ref(self, resource_version_ref: str):
+        def has_version_ref(p):
```

Comment:
is it really necessary to recursively process this?

Reply:
let me rephrase: please handle the different expected cases explicitly. I understood that there are a few explicit possible nesting variants. please handle those explicitly. please do _not_ process raw `dicts`, create model classes instead.